File: Java/JavaUtils/src/javautils/message/IHeartbeatMessage.java
package javautils.message;

import java.util.Date;

/*******************************************************************************
 * Heartbeat message interface.
 ******************************************************************************/
public interface IHeartbeatMessage extends IMessage
{
    /***************************************************************************
     * Returns the time this message was sent.
     * 
     * @return
     **************************************************************************/
    public Date getSendTime();

    /***************************************************************************
     * Returns the availability status of the user who sent this heartbeat.
     * 
     * @return
     **************************************************************************/
    public boolean isUserAvailable();

    /***************************************************************************
     * Returns the display name of the user who sent this heartbeat.
     * 
     * @return
     **************************************************************************/
    public String getUserDisplayName();
}

File: C#/ChatterBox/ChatterBox/IHeartbeatMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ChatterBox
{
    interface IHeartbeatMessage
    {
        /// <summary>
        /// Returns the time this message was sent.
        /// </summary>
        /// <returns></returns>
        DateTime getSendTime();

        /// <summary>
        /// Returns the availability status of the user who sent this message.
        /// </summary>
        /// <returns></returns>
        bool isUserAvailable();

        /// <summary>
        /// Returns the display name of this user who sent this message.
        /// </summary>
        /// <returns></returns>
        string getUserDisplayName();
    }
}
